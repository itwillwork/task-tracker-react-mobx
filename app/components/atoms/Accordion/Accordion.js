import React, { Component } from 'react';

import styles from './Accordion.scss';
import classNames from 'classnames/bind';
const cls = classNames.bind(styles);

class Accordion extends Component {
	constructor(props) {
		super(props);

		this.state = {
			showChildren: false,
		};

		this.showChildren = this.toggleShowChildren.bind(this, true);
		this.hideChildren = this.toggleShowChildren.bind(this, false);
	}
	toggleShowChildren(value) {
		this.setState({
			showChildren: value
		})
	}
	render() {

		if (!this.state.showChildren) {
			return (
				<div className={cls('Accordion')}>
					<span className={cls('toggle')} onClick={this.showChildren}>{this.props.textForShow}</span>
				</div>
			);
		}

		return (
			<div className={cls('Accordion')}>
				<div>
					<span className={cls('toggle')} onClick={this.hideChildren}>{this.props.textFotHide}</span>
				</div>
				<div className={cls('container')}>
					{this.props.children}
				</div>
			</div>
		);
	}
}

Accordion.defaultProps = {
	textForShow: 'Показать',
	textFotHide: 'Скрыть',
};

export default Accordion;