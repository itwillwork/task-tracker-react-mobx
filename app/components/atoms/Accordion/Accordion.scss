.Accordion {
  margin: 0 0 8px 0;
  .toggle {
    font-size: 12px;
    line-height: 15px;
    color: #0193d6;
    text-decoration: underline;
    cursor: pointer;
    &:hover {
      text-decoration: none;
    }
  }
  .container {
    margin: 8px 0;
  }
}