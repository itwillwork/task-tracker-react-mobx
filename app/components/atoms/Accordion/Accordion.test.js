// Link.react-test.js
import React from 'react';
import Accordion from './Accordion';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
	const component = renderer.create(
		<Accordion herf="http://www.facebook.com">Facebook</Accordion>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	component.getInstance().showChildren();
	// re-rendering
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	component.getInstance().hideChildren();
	// re-rendering
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});