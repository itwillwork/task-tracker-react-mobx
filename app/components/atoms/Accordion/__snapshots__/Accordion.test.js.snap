// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="Accordion"
>
  <span
    className="toggle"
    onClick={[Function]}
  >
    Показать
  </span>
</div>
`;

exports[`Link changes the class when hovered 2`] = `
<div
  className="Accordion"
>
  <div>
    <span
      className="toggle"
      onClick={[Function]}
    >
      Скрыть
    </span>
  </div>
  <div
    className="container"
  >
    Facebook
  </div>
</div>
`;

exports[`Link changes the class when hovered 3`] = `
<div
  className="Accordion"
>
  <span
    className="toggle"
    onClick={[Function]}
  >
    Показать
  </span>
</div>
`;
